format-common/src/main/java/io/cdap/cdap/format/io/BigDecimalAwareJsonStructuredRecordDatumWriter.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.format.io;
+
+import com.google.gson.stream.JsonWriter;
+import io.cdap.cdap.api.common.Bytes;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.common.io.Encoder;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+
+/**
+ * Custom Datum Writer for Structured Records, this class writes BigDecimal values as numbers instead of byte arrays in
+ * the JSON output.
+ */
+public class BigDecimalAwareJsonStructuredRecordDatumWriter extends JsonStructuredRecordDatumWriter {
+  @Override
+  protected void encode(Encoder encoder, Schema schema, Object value) throws IOException {
+    Schema nonNullableSchema = schema.isNullable() ? schema.getNonNullable() : schema;
+    Schema.LogicalType logicalType = nonNullableSchema.getLogicalType();
+
+    if (value != null && logicalType == Schema.LogicalType.DECIMAL) {
+      BigDecimal bdValue = fromObject(value, schema);
```

Comment:
BTW: Can we have a BigDecimalWrapper right here? This would remove the need for a special JsonWriter

Reply:
We still to call the writer directly, as the superclass doesn't expose methods to write

Reply:
I understand why do you use it directly (and that another question above, we should just expose it). What I am custion here is it you do wrapping right here, you would not need `BigDecimalAwareJsonWriter` and use just vanilla `JsonWriter`, no?

Reply:
The problem is the JsonWriter's  `value(Number number)` method uses the Number class `toString()` method internally to generate the representation of the value that it needs to output.

The problem with this is that BigQuery's `toString()` method can represent values in scientific notation. In order to avoid this, I create that wrapper which delegates to BigQuery's `toPlainString()` method.

Reply:
Yep. If you pass this Wrapper to vanilla `JsonWriter.value(Number)` it would do the job, no? You don't need to create wrapper inside `JsonWriter`